static/app/components/modals/dashboardWidgetQuerySelectorModal.tsx

```diff
@@ -38,6 +38,9 @@ class DashboardWidgetQuerySelectorModal extends React.Component<Props> {
         selection,
         widget.displayType
       );
+      const discoverLocation = eventView.getResultsViewUrlTarget(organization.slug);
+      // Pull Y-Axis from the widget
+      discoverLocation.query.yAxis = query.fields;
```

Comment:
Is it possible for `query.fields` to contain yAxis incompatible fields here?

Reply:
Good call, I've added some checks to ensure we only use compatible fields via `eventView.getYAxisOptions` and also made sure to limit to 3 yAxis max as the table widget allows 3+ fields to be selected..